misc/db-updates/2023-07-17/creatorsAsNonClassicDataObjects

```diff
@@ -0,0 +1,71 @@
+#!/usr/local/bin/php -d mysqlnd.net_read_timeout=86400
+<?php
+set_include_path("../../../include");
+require("header.inc.php");
+
+$shardHostID = $argv[1];
+$startShard = $argv[2];
+$stopShard = $argv[3];
+
+Z_Core::$debug = true;
+
+$shardIDs = Zotero_DB::columnQuery("SELECT shardID FROM shards WHERE shardHostID=? AND shardID >= ? AND shardID <= ? ORDER BY shardID", [$shardHostID, $startShard, $stopShard]);
+foreach ($shardIDs as $shardID) {
+	echo "Shard: $shardID\n";
+
+	echo "Setting shard to readonly\n";
+	Zotero_DB::query("UPDATE shards SET state='readonly' WHERE shardID=?", $shardID);
+	
+	echo "Waiting 60 seconds for requests to stop\n";
+	sleep(60);
+	
+	// Creators 
+	echo "Migrating creators\n";
+	// Drop foreign key constraint
+	Zotero_Admin_DB::query("ALTER TABLE `itemCreators` DROP CONSTRAINT `itemCreators_ibfk_1`;", false, $shardID);
+	Zotero_Admin_DB::query("ALTER TABLE `itemCreators` DROP CONSTRAINT `itemCreators_ibfk_2`;", false, $shardID);
+
+	// Create new itemCreators table
+	Zotero_Admin_DB::query("CREATE TABLE `itemCreatorsNew` ( `creatorID` BIGINT UNSIGNED NOT NULL, `itemID` BIGINT UNSIGNED NOT NULL, `firstName` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci DEFAULT NULL, `lastName` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci DEFAULT NULL, `fieldMode` tinyint(1) UNSIGNED DEFAULT NULL, `creatorTypeID` smallint(5) UNSIGNED NOT NULL, `orderIndex` smallint(5) UNSIGNED NOT NULL, PRIMARY KEY (`creatorID`, `itemID`), KEY `creatorTypeID` (`creatorTypeID`), KEY `name` (`lastName`(7),`firstName`(6)) ) ENGINE=InnoDB DEFAULT CHARSET=utf8;", false, $shardID);
```

Comment:
Sorry, I led you astray originally, and didn't clarify later. I originally said there'd still be `creatorID` in `creators`, but since we're inlining the data in `itemCreators`, there's no reason for there to be a `creatorID` at all — it's not serving any purpose. And the primary key should be on `itemID, orderIndex`, since that's what's unique. (It was actually wrong to begin with — it should've been on `itemID, orderIndex` before like in the client, not `itemID, creatorID, orderIndex`.)